### ISSUE TITLE
A unique combination of flags leads to an error

### ISSUE THREAD

**Issue by pirj** [rubocop-hq/rubocop#8465]
```
rubocop --debug --format simple --only RSpec/MultipleMemoizedHelpers ~/source/rspec-dev/repos/rspec-rails
```

No other commands lead to the same result:
 - without `--format simple` it's ok
 - with a different path it's ok
 - with a different target repository it's ok

Other formatters, however, yield the same result, tried `quiet`, `progress`, `tap`, `pacman`.

`multiple_memoized_helpers.rb:101:in '<='` is:
```
          return if count <= max
```
and `max` is defined as:
```ruby
        def max
          cop_config['Max']
        end
```

Setting a breakpoint just before the failure:
```
binding.pry if max.nil?
```
shows:
```
[5] pry(#<RuboCop::Cop::RSpec::MultipleMemoizedHelpers>)> cop_config
=> {"Enabled"=>true}
```
It is the same on the very first entry to `on_block`.

which is quite odd on its own since in `config/default.yml` it's `Enabled: false`:

```yaml
RSpec/MultipleMemoizedHelpers:
  Description: Checks if example groups contain too many `let` and `subject` calls.
  Enabled: false
  AllowSubject: false
  Max: 5
  StyleGuide: https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/MultipleMemoizedHelpers
```

It's understandable that to run it as `--only` we override `Enable`, but it seems in some cases other settings are getting wiped out.

--------

## Expected behavior

No errors

## Actual behavior

```
An error occurred while RSpec/MultipleMemoizedHelpers cop was inspecting /Users/pirj/source/rspec-dev/repos/rspec-rails/spec/rspec/rails/example/helper_example_group_spec.rb:59:2.
comparison of Integer with nil failed
/Users/pirj/source/rubocop-rspec/lib/rubocop/cop/rspec/multiple_memoized_helpers.rb:101:in `<='
/Users/pirj/source/rubocop-rspec/lib/rubocop/cop/rspec/multiple_memoized_helpers.rb:101:in `on_block'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/cop/commissioner.rb:91:in `block (2 levels) in trigger_responding_cops'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/cop/commissioner.rb:112:in `with_cop_error_handling'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/cop/commissioner.rb:90:in `block in trigger_responding_cops'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/cop/commissioner.rb:89:in `each'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/cop/commissioner.rb:89:in `trigger_responding_cops'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/cop/commissioner.rb:61:in `block (2 levels) in <class:Commissioner>'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-ast-0.3.0/lib/rubocop/ast/traversal.rb:59:in `block in on_begin'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-ast-0.3.0/lib/rubocop/ast/traversal.rb:59:in `each'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-ast-0.3.0/lib/rubocop/ast/traversal.rb:59:in `on_begin'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/cop/commissioner.rb:62:in `block (2 levels) in <class:Commissioner>'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-ast-0.3.0/lib/rubocop/ast/traversal.rb:159:in `on_while'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/cop/commissioner.rb:62:in `block (2 levels) in <class:Commissioner>'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-ast-0.3.0/lib/rubocop/ast/traversal.rb:14:in `walk'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/cop/commissioner.rb:74:in `investigate'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/cop/team.rb:151:in `investigate_partial'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/cop/team.rb:83:in `investigate'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/runner.rb:295:in `inspect_file'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/runner.rb:245:in `block in do_inspection_loop'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/runner.rb:277:in `block in iterate_until_no_changes'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/runner.rb:270:in `loop'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/runner.rb:270:in `iterate_until_no_changes'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/runner.rb:241:in `do_inspection_loop'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/runner.rb:121:in `block in file_offenses'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/runner.rb:146:in `file_offense_cache'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/runner.rb:120:in `file_offenses'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/runner.rb:111:in `process_file'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/runner.rb:90:in `block in each_inspected_file'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/runner.rb:89:in `each'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/runner.rb:89:in `reduce'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/runner.rb:89:in `each_inspected_file'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/runner.rb:78:in `inspect_files'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/runner.rb:39:in `run'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/cli/command/execute_runner.rb:21:in `execute_runner'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/cli/command/execute_runner.rb:13:in `run'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/cli/command.rb:10:in `run'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/cli/environment.rb:17:in `run'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/cli.rb:65:in `run_command'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/cli.rb:72:in `execute_runners'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/lib/rubocop/cli.rb:41:in `run'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/exe/rubocop:13:in `block in <top (required)>'
/Users/pirj/.rvm/rubies/ruby-2.5.1/lib/ruby/2.5.0/benchmark.rb:308:in `realtime'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/gems/rubocop-0.89.0/exe/rubocop:12:in `<top (required)>'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/bin/rubocop:23:in `load'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/bin/rubocop:23:in `<main>'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/bin/ruby_executable_hooks:24:in `eval'
/Users/pirj/.rvm/gems/ruby-2.5.1@rubocop-rspec/bin/ruby_executable_hooks:24:in `<main>'
```

## Steps to reproduce the problem

Checkout https://github.com/rubocop-hq/rubocop-rspec/pull/863
Checkout https://github.com/rspec/rspec-rails/
From `rubocop-rspec`, run:
```
rubocop --debug --format simple --only RSpec/MultipleMemoizedHelpers ~/source/rspec-rails
```

## RuboCop version

```
$ rubocop -V
0.89.0 (using Parser 2.7.1.4, rubocop-ast 0.3.0, running on ruby 2.5.1 x86_64-darwin18)
```

**Comment by pirj** [comment]
I'll close this, as it never happened to me ever since, and is an edge case.
